server/src/api/event_filter.go

```diff
@@ -0,0 +1,238 @@
+package api
+
+import (
+	"encoding/json"
+	"sort"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/websocket"
+	dto2 "scrumlr.io/server/common/dto"
+	"scrumlr.io/server/database/types"
+	"scrumlr.io/server/logger"
+	"scrumlr.io/server/realtime"
+)
+
+func isModerator(clientID uuid.UUID, sessions []*dto2.BoardSession) bool {
+	for _, session := range sessions {
+		if clientID == session.User.ID {
+			if session.Role == types.SessionRoleModerator || session.Role == types.SessionRoleOwner {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// ToDo: Refactor the client sorting or even find a better solution to avoid it
+type client struct {
+	userID uuid.UUID
+	conn   *websocket.Conn
+	role   types.SessionRole
+}
+
+type ByRole []client
+
+func (b ByRole) Len() int {
+	return len(b)
+}
+
+func (b ByRole) Less(i, j int) bool {
+	return b[i].role < b[j].role
+}
+
+func (b ByRole) Swap(i, j int) {
+	b[i], b[j] = b[j], b[i]
+}
+
+func sortClientsByRole(clients map[uuid.UUID]*websocket.Conn, boardParticipants []*dto2.BoardSession) []client {
```

Comment:
why exactly does it need to be sorted? I can't figure it out 😅

Reply:
For some reason, when the clients are not sorted, an unintended behavior occurs so that, at random, cards get deleted from the "participants" board. I still don't understand why this is happening...

Reply:
alright, that's ... interesting to say the least 👀   
could you put a comment then so future scrumlr people know why this is required? :)

Reply:
I added a comment for now, but will also take a second look – now that I have worked on filtering for quite a while. Maybe the issue can be solved directly.

Reply:
resolved.